crates/bindings-csharp/BSATN.Codegen/Utils.cs

```diff
@@ -126,25 +113,19 @@ static string GetTypeInfoForNamedType(INamedTypeSymbol type)
                 }
                 return $"SpacetimeDB.BSATN.Enum<{SymbolToName(type)}>";
             }
-            var result = type.OriginalDefinition.ToString() switch
+            var fullName = type.ToDisplayString(SymbolFormatForBSATN);
+            string result;
+            if (fullName.StartsWith("System."))
```

Comment:
Can't you compare against type.Namespace here?

Reply:
`.ContainingNamespace` would have the full namespace, eg `System.Collections.Generic`. You'd then need to loop through parents to get to the actual top. That's what I did initially, but since we need this stringified name anyway, it's much simpler to do a string-based check as well.